lib/SILOptimizer/Utils/InstOptUtils.cpp

```diff
@@ -171,9 +172,273 @@ bool swift::isIntermediateRelease(SILInstruction *inst,
   return false;
 }
 
+static bool hasOnlyEndOfScopeOrDestroyUses(SILInstruction *inst) {
```

Comment:
At `-O` make sure the existence of DebugValue instructions cannot change the output of the compiler... i.e. they should only prevent dead instruction deletion at `-Onone`.